labgrid/driver/powerdriver.py

```diff
@@ -187,3 +188,70 @@ def cycle(self):
     @Driver.check_active
     def get(self):
         return self.pykush.get_port_state(self.port.index)
+
+@target_factory.reg_driver
+@attr.s(cmp=False)
+class SerialPortPowerDriver(Driver, PowerProtocol):
+    """
+    Controls the power of a DUT using the control lines of a serial port.
+
+    This driver uses the flow-control pins of a serial port (for example
+    an USB-UART-dongle) to control some external power switch. You may connect
+    some kind of relay board to the flow control pins.
+
+    The serial port should NOT be used for serial communication at the same
+    time. This will probably reset the flow-control signals.
+
+    Usable signals are DTR and RTS.
+    """
+
+    port = attr.ib(validator=attr.validators.instance_of(str))
+    signal= attr.ib(validator=attr.validators.instance_of(str))
```

Comment:
IMO extending the existing resources with an optional `power_signal` ist more straightforward.